argoverse/evaluation/stereo/model_analysis.py

```diff
@@ -0,0 +1,149 @@
+        f"Mean forward time (ms): {time_stats['mean_time_ms']} \n"
+        f"Std forward time (ms): {time_stats['std_time_ms']} \n"
+        f"{input_size_str} \n"
+        f"{count_table}"
+    )
+    print(output_str)
+
+    report_fpath = output_dir / "model_analysis_report.txt"
+    with open(report_fpath, "w") as f:
+        print(output_str, file=f)
+
+    print(f"Saved report at {report_fpath}.")
+
+
+def compute_forward_time(
+    model: torch.nn.Module,
+    inputs: Union[torch.Tensor, Tuple[torch.Tensor, ...]],
+    num_forward_passes: int = 30,
+) -> Dict[str, float]:
+    """Compute the model forward time.
+
+    Args:
+        model: The model to perform the analysis on.
+        inputs: Inputs that are passed to the model.
+        num_forward_passes: Number of forward passes to compute the time statistics.
+
+    Returns:
+        A dictionary that records the time statistics (mean and std).
+    """
+    # Init time loggers
+    start = torch.cuda.Event(enable_timing=True)
+    end = torch.cuda.Event(enable_timing=True)
+
+    # Before we run the profiler, we warm-up CUDA to ensure accurate performance benchmarking
+    model(*inputs)
+
+    # Model forward passes
+    model.eval()
+    timings_ms = []
+    with torch.no_grad():
+        for _ in track(range(num_forward_passes), description="Forward passing..."):
+            start.record()
+            model(*inputs)
+            end.record()
+
+            # Wait for GPU synchronization
+            torch.cuda.synchronize()
+
+            time_ms = start.elapsed_time(end)
+            timings_ms.append(time_ms)
+
+    # Compute mean time and its standard deviation
+    mean_time_ms = np.sum(timings_ms) / num_forward_passes
+    std_time_ms = np.std(timings_ms)
+
+    return {"mean_time_ms": mean_time_ms, "std_time_ms": std_time_ms}
+
+
+if __name__ == "__main__":
+
+    # Define dummy stereo model
+    class DummyStereoModel(torch.nn.Module):  # type: ignore
+        def __init__(self) -> None:
+            super().__init__()
+            self.conv_layer1 = torch.nn.Conv2d(in_channels=3, out_channels=16, kernel_size=3)
+            self.conv_layer2 = torch.nn.Conv2d(in_channels=32, out_channels=3, kernel_size=3)
+
+        def forward(self, input1: torch.Tensor, input2: torch.Tensor) -> torch.Tensor:
+            output1 = self.conv_layer1(input1)
+            output2 = self.conv_layer1(input2)
+            output = self.conv_layer2(torch.cat((output1, output2), axis=1))
+            return output
+
+    model = DummyStereoModel().cuda()
+    device = next(model.parameters()).device
+    dtype = next(model.parameters()).dtype
+
+    # Define dummy inputs
+    input_shape = (1, 3, 10, 10)  # E.g., batch_size x num_channels x height x width
+    input1 = torch.randn(input_shape, device=device, dtype=dtype)
+    input2 = torch.randn(input_shape, device=device, dtype=dtype)
+    dummy_input = (input1, input2)
+
+    report_dir = Path("/home/ubuntu/")
```

Comment:
```suggestion
    report_dir = Path.home()
```
Automatic detection of their home directory.